### PR TITLE
Add tests for all padding layers to `module_db` in `common_modules.py`

### PR DESCRIPTION
**Pull request by rishub-tamirisa**
Adding the PR discussed in #96295.

- Adds tests for all current padding layers to `module_db` in `torch/testing/_internal/common_modules.py` ( `nn.ReflectionPad`, `nn.ReplicationPad`, `nn.ZeroPad`, `nn.ConstantPad` ) for 1D, 2D, and 3D variants.
- Removes tests for the same padding layers from `torch/testing/_internal/common_nn.py`

cc @soulitzer @jbschlosser @albanD

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
This is great! Thanks a lot for helping with this!
The simplest way to land this PR is going to add the proper skips for all the failing tests and then we can look into fixing relevant ones in follow ups.
In particular:
- You can skip from common_modules.py the memory format and autograd failures.
- You can skip the test_aotdispatch.py failures directly there (you'll see there are a lot of other tests skipped already)

Don't hesitate to ask if you need help!

**Comment by rishub-tamirisa** [comment]
@albanD Thanks for the quick review - I'll add the skips in the next commit. Just checking, do any of these padding operations require a `reference_fn`? Given that they use `torch.nn.functional.pad` under the hood, does it make sense for me to be testing correctness here?

**Comment by albanD** [comment]
If you can easily write one with something like `numpy.pad` then I think it is worth adding.
But for Modules with so many subtle parameters, it can be challenging to write a reference, so it is ok if you don't want to do it as well.

**Review by albanD — APPROVED** [review]
Looks great!
Thanks a lot. I opened issues to investigate the failures. Feel free to comment there if you want to look into that more!

@pytorchbot merge

**Comment by rishub-tamirisa** [comment]
@albanD I don't think I have permissions to add a label.

Also, I'm definitely interested in looking deeper into these failures, I'll comment there if I have something more substantive to contribute on that!

**Comment by albanD** [comment]
In the future, you can also ask the bot to add labels for you!
@pytorchbot -h
